### PR TITLE
Initial CORS implementation

### PR DESCRIPTION
**Pull request by h2non**
Pretty small package. Just does the job.

### CODE REVIEW

File: cors_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	})
+
+	res := httptest.NewRecorder()
+	req, _ := http.NewRequest("GET", "http://example.com/foo", nil)
+
+	s(testHandler).ServeHTTP(res, req)
+
+	assertHeaders(t, res.Header(), map[string]string{
+		"Vary": "Origin",
+		"Access-Control-Allow-Origin":      "",
```
Comment (groyoh) [inline cors_test.go:34]: Shouldn't this be `"*"` by default?

**Comment by groyoh** [comment]
The code looks good to me. Would like to see more tests though :wink:

**Comment by groyoh** [comment]
Also is travis setup?

File: cors_test.go (hunk 1 above)
Comment (h2non) [inline cors_test.go:34]: No, unless you explicitly want it. It's important in terms of security.

**Comment by h2non** [comment]
Yes, Travis is active now.

File: cors.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+)
+
+// Options is a configuration container to setup the CORS middleware.
+type Options struct {
+	// AllowedOrigins is a list of origins a cross-domain request can be executed from.
+	// If the special "*" value is present in the list, all origins will be allowed.
+	// An origin may contain a wildcard (*) to replace 0 or more characters
+	// (i.e.: http://*.domain.com). Usage of wildcards implies a small performance penality.
+	// Only one wildcard can be used per origin.
+	// Default value is ["*"]
```
Comment (groyoh) [inline cors.go:18]: Based on https://github.com/vinxi/cors/pull/1/files#discussion_r59881747, I'd say that this doc is not correct.

Comment (h2non) [inline cors.go:18]: Seems like if you don't define any `AllowedOrigins`, the default behavior will be allow any origin, which is analog to `*`. I was wrong about this. You can always define `AllowedOrigins: ["*"]` if you want it explicitly.

**Pull request merged**